Sub-configs stopped supporting json files

**What version of ESLint are you using?**
2.0.0

**What configuration and parser (Espree, Babel-ESLint, etc.) are you using?**
Babel-ESLint

**What did you do? Please include the actual source code causing the issue.**
Updated eslint from v1 to v2 with my shareable config.

**What did you expect to happen?**
My shareable config would continue to work with sub-configs as json files.

**What actually happened? Please include the actual, raw output from ESLint.**
```
D:\Projects\project\node_modules\eslint\lib\config\config-file.js:361
            throw e;
            ^

Error: Cannot find module 'eslint-config-noamokman/test' from 'D:\Projects\project\node_modules'
Referenced from: D:\Projects\project\test\.eslintrc
    at Function.module.exports [as sync] (D:\Projects\project\node_modules\resolve\lib\sync.js:33:11)
    at resolve (D:\Projects\project\node_modules\eslint\lib\config\config-file.js:417:38)
    at load (D:\Projects\project\node_modules\eslint\lib\config\config-file.js:436:24)
    at D:\Projects\project\node_modules\eslint\lib\config\config-file.js:355:36
    at Array.reduceRight (native)
    at applyExtends (D:\Projects\project\node_modules\eslint\lib\config\config-file.js:338:28)
    at Object.load (D:\Projects\project\node_modules\eslint\lib\config\config-file.js:464:22)
    at loadConfig (D:\Projects\project\node_modules\eslint\lib\config.js:67:33)
    at getLocalConfig (D:\Projects\project\node_modules\eslint\lib\config.js:129:23)
    at Config.getConfig (D:\Projects\project\node_modules\eslint\lib\config.js:222:22)
```

Hi, now that we got the formalities out of the way,
I have a shareable config called [eslint-config-noamokman](https://github.com/noamokman/eslint-config-noamokman).
I recently updated it to work with eslint 2.
But it seems that the `test` sub config stopped working.
After checking the problem, it seems that the resolve module can't search for a json file when given: `eslint-config-noamokman/test`.

My module has an `eslint.json` for the main config, and a `test.json` for testing.
Changing from `test.json` to `test.js` fixes the problem, but I would like to add support for both.

I know that I have to open an issue first, so if everything looks good I would like to submit a PR.

Thanks,
Noam.

--- a/lib/config/config-file.js
+++ b/lib/config/config-file.js
@@ -409,12 +409,14 @@ function resolve(filePath, relativeTo) {
             var packagePath = filePath.substr(7, filePath.lastIndexOf("/") - 7);
             var configName = filePath.substr(filePath.lastIndexOf("/") + 1, filePath.length - filePath.lastIndexOf("/") - 1);
             filePath = resolveModule.sync(normalizePackageName(packagePath, "eslint-plugin"), {
-                basedir: getLookupPath(relativeTo)
+                basedir: getLookupPath(relativeTo),
+                extensions: [".js", ".json"]
             });
             return { filePath: filePath, configName: configName };
         } else {
             filePath = resolveModule.sync(normalizePackageName(filePath, "eslint-config"), {
-                basedir: getLookupPath(relativeTo)
+                basedir: getLookupPath(relativeTo),
+                extensions: [".js", ".json"]
             });
             return { filePath: filePath };
         }

--- a/tests/lib/config/config-file.js
+++ b/tests/lib/config/config-file.js
@@ -720,7 +720,11 @@ describe("ConfigFile", function() {
                 [ "eslint-configfoo", getRelativeModulePath("eslint-config-eslint-configfoo", relativePath), relativePath],
                 [ "@foo/eslint-config", getRelativeModulePath("@foo/eslint-config", relativePath), relativePath],
                 [ "@foo/bar", getRelativeModulePath("@foo/eslint-config-bar", relativePath), relativePath],
-                [ "plugin:@foo/bar/baz", getRelativeModulePath("@foo/eslint-plugin-bar", relativePath), relativePath]
+                [ "plugin:@foo/bar/baz", getRelativeModulePath("@foo/eslint-plugin-bar", relativePath), relativePath],
+                [ "eslint-config-foo/bar", path.resolve("./node_modules", "eslint-config-foo/bar", "index.json"), relativePath],
+                [ "eslint-config-foo/bar", path.resolve("./node_modules", "eslint-config-foo", "bar.json"), relativePath],
+                [ "eslint-config-foo/bar", path.resolve("./node_modules", "eslint-config-foo/bar", "index.js"), relativePath],
+                [ "eslint-config-foo/bar", path.resolve("./node_modules", "eslint-config-foo", "bar.js"), relativePath]
             ], function(input, expected, relativeTo) {
                 it("should return " + expected + " when passed " + input, function() {
 

